### PR TITLE
Combined docs with CRT

### PR DESCRIPTION
**Pull request by justinboswell**
Depends on: 
* https://github.com/awslabs/aws-crt-nodejs/pull/89

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: make-docs.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
```
Comment (justinboswell) [inline make-docs.sh:1]: This is the script that actually builds the docs

File: package.json [hunk 2]
```diff
@@ -23,14 +23,12 @@
     "build": "tsc"
   },
   "devDependencies": {
+    "@convergencelabs/typedoc-plugin-custom-modules": "^0.4.0",
```
Comment (justinboswell) [inline package.json:26]: This plugin allows the @module and @moduledefinition tags to work

File: tdconfig.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (justinboswell) [inline tdconfig.json:1]: This is a minimal tsconfig.json for typescript that can be used to generate the docs

File: lib/index.ts [hunk 4]
```diff
@@ -12,14 +12,21 @@
 * permissions and limitations under the License.
 */
 
+/**
+ * @moduledefinition aws-iot-device-sdk
```
Comment (justinboswell) [inline lib/index.ts:16]: This creates an aws-iot-device-sdk module in the docs

File: tdconfig.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+            "dom"
+        ],
+        "allowSyntheticDefaultImports": true
+    },
+    "include": [
+        "lib/index.ts",
+        "lib/**/*.ts",
+        "build/docs/aws-crt-nodejs/lib/index.ts",
+        "build/docs/aws-crt-nodejs/lib/common/*.ts",
+        "build/docs/aws-crt-nodejs/lib/native/*.ts"
```
Comment (DavidOgunsAWS) [inline tdconfig.json:15]: Browser is intentionally missing here?

Comment (justinboswell) [reply tdconfig.json:15]: Yes. We'll bifurcate later, but for now, just keep it consistent with just the node stuff.

**Pull request merged**